datatypes-jws/src/jvmTest/kotlin/JsonWebKeyJvmTest.kt

```diff
@@ -1,69 +1,106 @@
+import at.asitplus.crypto.datatypes.asn1.encodeToByteArray
 import at.asitplus.crypto.datatypes.asn1.ensureSize
 import at.asitplus.crypto.datatypes.jws.JsonWebKey
+import at.asitplus.crypto.datatypes.jws.io.jsonSerializer
 import at.asitplus.crypto.datatypes.jws.toJsonWebKey
 import io.kotest.core.spec.style.FreeSpec
 import io.kotest.matchers.nulls.shouldNotBeNull
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.string.shouldHaveMinLength
+import kotlinx.serialization.serializer
 import java.security.KeyPair
 import java.security.KeyPairGenerator
 import java.security.interfaces.ECPublicKey
+import java.security.interfaces.RSAPublicKey
 
 class JsonWebKeyJvmTest : FreeSpec({
 
     lateinit var ecCurve: EcCurve
     lateinit var keyPair: KeyPair
+    lateinit var keyPairRSA: KeyPair
 
     beforeTest {
         ecCurve = EcCurve.SECP_256_R_1
-        keyPair = KeyPairGenerator.getInstance("EC").also {
-            it.initialize(256)
-        }.genKeyPair()
+        keyPair = KeyPairGenerator.getInstance("EC").also { it.initialize(256) }.genKeyPair()
+        keyPairRSA = KeyPairGenerator.getInstance("RSA").also { it.initialize(2048) }.genKeyPair()
     }
 
     "JWK can be created from Coordinates" - {
         val xFromBc = (keyPair.public as ECPublicKey).w.affineX.toByteArray().ensureSize(ecCurve.coordinateLengthBytes)
         val yFromBc = (keyPair.public as ECPublicKey).w.affineY.toByteArray().ensureSize(ecCurve.coordinateLengthBytes)
-        val jsonWebKey = CryptoPublicKey.Ec.fromCoordinates(ecCurve, xFromBc, yFromBc).toJsonWebKey()
+        val pubKey = CryptoPublicKey.Ec.fromCoordinates(ecCurve, xFromBc, yFromBc)
+        val jsonWebKey = pubKey.toJsonWebKey()
 
         jsonWebKey.shouldNotBeNull()
         jsonWebKey.x shouldBe xFromBc
         jsonWebKey.y shouldBe yFromBc
         jsonWebKey.keyId.shouldNotBeNull()
         jsonWebKey.keyId shouldHaveMinLength 32
 
-        "it can be recreated" {
+        "it can be recreated from keyId" {
             val recreatedJwk = JsonWebKey.fromKeyId(jsonWebKey.keyId!!)
             recreatedJwk.shouldNotBeNull()
             recreatedJwk.keyId shouldBe jsonWebKey.keyId
             recreatedJwk.x shouldBe jsonWebKey.x
             recreatedJwk.y shouldBe jsonWebKey.y
         }
+
+        "it can be converted back to CryptoPublicKey" {
+            val recreatedPubKey = jsonWebKey.toCryptoPublicKey()
+            (pubKey == recreatedPubKey) shouldBe true
+        }
     }
 
-    "JWK can be created from ANSI X962" - {
-        val xFromBc = (keyPair.public as ECPublicKey).w.affineX.toByteArray().ensureSize(ecCurve.coordinateLengthBytes)
-        val yFromBc = (keyPair.public as ECPublicKey).w.affineY.toByteArray().ensureSize(ecCurve.coordinateLengthBytes)
-        val ansiX962 = byteArrayOf(0x04) + xFromBc + yFromBc
-        val jsonWebKey = CryptoPublicKey.Ec.fromAnsiX963Bytes(ansiX962)!!.toJsonWebKey()
+    //Todo move to CryptoPublicKeyTest
```

Comment:
TODO

Reply:
Test case targets deprecated function. Will be removed when VcLib uses the new functions. Deleting test case